src/emoji-picker/emoji-state.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+export type EmojiState = {
+  isPickerOpen?: boolean;
+};
+const initialState: EmojiState = { isPickerOpen: false };
+
+const TOGGLE_EMOJI_PICKER = 'toggleEmojiPicker';
+
+type ToggleEmojiPicker = {
+  type: typeof TOGGLE_EMOJI_PICKER;
+  isPickerOpen: boolean;
+};
+
+export const toggleEmojiPicker = (): ToggleEmojiPicker => ({
+  type: TOGGLE_EMOJI_PICKER,
+  isPickerOpen: false,
+});
```

Comment:
I think we don't need the property `isPickerOpen` on the `toggleEmojiPicker` action.
The reducer only looks for the **current state** to decide the next value for `isPickerOpen`.